### PR TITLE
Update of Phase2 Tracker Digitizer to fix an issue in Premixing

### PR DESCRIPTION
**Pull request by suchandradutta**
#### PR description:
Updating Phase2TrackerDigitizerAlgorithm to fix the issue observed during Premixing of PU events at the Digitization. It was observed that the Digitized signal gets lowered and correspondingly #  Digis are lower than expected in inner Pixel layers . It was found that this is due to the fact that the Dual-Slope signal scaling is used. It is now fixed using simple liner scaling of charge during the first step  of Pre-Mixing.  

  - Changes are NOT expected! As shown in the validation plots, the PreMixing and noPreMixing show similar behaviour as expected.  
  - NO other PRs or externals it depends upon it
  - The details of the code change and the validation results are reported to Tk Phase2 Simulation group [Link](https://indico.cern.ch/event/1321219/contributions/5602295/attachments/2726551/4738571/Phase2TrackerDigi_PreMixing_Validation_29092023.pdf)
  
@emiglior

### CODE REVIEW

**Comment by srimanob** [comment]
test parameters:
- workflow = 25034.999

**Comment by srimanob** [comment]
@cmsbuild please test

File: SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py [hunk 1]
```diff
@@ -233,11 +233,17 @@
         AddNoisyPixels = False,
         AddInefficiency = False,
         AddThresholdSmearing = False,
+        Phase2ReadoutMode = -1,
+        AdcFullScale = 255,
+        AddXTalk = False,
```
Comment (mmusich) [inline SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py:238]: I would recommend documenting this here:
https://github.com/cms-sw/cmssw/blob/f738feebb2f2a520baac4e21cd3da20e3e52236e/SimTracker/SiPhase2Digitizer/python/phase2TrackerDigitizer_cfi.py#L219-L223

**Comment by srimanob** [comment]
Hi @suchandradutta 
Could you please confirm that DQM plot shown by PR test is as expected? For example, DigiOccupancyVsEtaP of Barrel IT [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-10-03-1100+e80085/59122/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR/TrackerPhase2ITDigi_DigiMonitor_Barrel_Layer1.html).

I assume the change is expected, only Phase-2 PreMix is changed with this PR.

**Comment by suchandradutta** [comment]
> Hi @suchandradutta Could you please confirm that DQM plot shown by PR test is as expected? For example, DigiOccupancyVsEtaP of Barrel IT [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-10-03-1100+e80085/59122/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR/TrackerPhase2ITDigi_DigiMonitor_Barrel_Layer1.html).
> 
> I assume the change is expected, only Phase-2 PreMix is changed with this PR.

@srimanob can you please let me know the workflow number where the difference is seen? So that I can re-run and check.

**Comment by srimanob** [comment]
> > Hi @suchandradutta Could you please confirm that DQM plot shown by PR test is as expected? For example, DigiOccupancyVsEtaP of Barrel IT [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-10-03-1100+e80085/59122/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR/TrackerPhase2ITDigi_DigiMonitor_Barrel_Layer1.html).
> > I assume the change is expected, only Phase-2 PreMix is changed with this PR.
> 
> @srimanob can you please let me know the workflow number where the difference is seen? So that I can re-run and check.

25034.999

**Comment by suchandradutta** [comment]
> Hi @suchandradutta Could you please confirm that DQM plot shown by PR test is as expected? For example, DigiOccupancyVsEtaP of Barrel IT [here](https://cmssdt.cern.ch/SDT/jenkins-artifacts/baseLineComparisons/CMSSW_13_3_X_2023-10-03-1100+e80085/59122/25034.999_TTbar_14TeV+2026D98PU_PMXS1S2PR/TrackerPhase2ITDigi_DigiMonitor_Barrel_Layer1.html).
> 
> I assume the change is expected, only Phase-2 PreMix is changed with this PR.

@srimanob I can fully confirm the changes observed in Inner Tracker Digi Occupancy/Charge histograms exactly as expected. I have updated the configuration file with some comments and switched off x-talk simulation also fo outer tracker layers in first step of PreMixing. Test were also done with the updated configuration file.

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by civanch** [comment]
please test

**Comment by civanch** [comment]
+1

only phase2 WFs are affected, which is expected

**Comment by srimanob** [comment]
+Upgrade

Change on Phase-2 premix workflow is expected.
@civanch do you want the update to present at FullSim meeting?

**Comment by civanch** [comment]
yes, we may discuss this Friday.

**Comment by antoniovilela** [comment]
+1

**Pull request merged**